app/components/Form/SubmitButtons.tsx

```diff
@@ -82,7 +92,7 @@ const SubmitButtons = ({
       <Button variant="primary" disabled={disabled}>
         {formatSubmitBtn()}
       </Button>
-      {isSubmitPage && isDraft && (
+      {isDraftAndSubmitPage && (
         <>
```

Comment:
I think we can get rid of these redundant fragments surrounding this button other than that this is looks good to me!